(WIP)deviseの導入

rinanada opened the pull request:
# WHAT

素材のダウンロード、gem deviseの導入、userモデルの作成、usersコントローラーの作成（test抜き)
# WHY

ログイン機能設定のため

Commit:
素材のダウンロード、deviseの導入、userモデルの作成、usersコントローラーの作成（test抜き)

yusukeyamane commented on app/views/users/index.html.haml:
```diff
@@ -0,0 +1,156 @@
+.container.user-page
```
userにindexは今回は必要無さそうです！

yusukeyamane commented on test/fixtures/users.yml:
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
今回もtestファイルは要らないです！

yusukeyamane commented on test/models/user_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
今回もtestファイルは要らないです！

Commit:
testファイル、indexファイルを取り除く

seoky0 commented on app/views/devise/sessions/new.html.haml:
```diff
@@ -0,0 +1,16 @@
+  .text-center
+    %p
+      Not a member?
+      %a{href: "/users/sign_in"} Sign up now
```
%aをlink_toで書き直してみましょう！
ここ以外にも%aがたくさんあるので、すべてlink_toに書き直しましょう^^

seoky0 commented on db/migrate/20160716034123_devise_create_users.rb:
```diff
@@ -0,0 +1,42 @@
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+
```
userのカラムは他にも必要なものがありそうです！

Commit:
%a →　=link_to, nickname/password/pro_imgカラムの追加

seoky0 commented on app/views/devise/sessions/new.html.haml:
```diff
@@ -13,4 +13,4 @@
     %p
       Not a member?
-      %a{href: "/users/sign_in"} Sign up now
+      = link_to{href: "/users/sign_in"} Sign up now
```
この記法でちゃんとviewに表示されていますか？

seoky0 commented on db/schema.rb:
```diff
@@ -11,7 +11,7 @@
 #
```
aタグをlink_toに書き換えることと、カラムの編集は全く別の実装内容なのでコミットを分けたほうが良いですね！
今回で言うとaタグをlink_toに書き換えたで一つのコミット、
カラムの編集で一つのコミットのように分けましょう！
一つの実装ごとにコミットするようにしましょう^^

Commit:
ログイン機能　未ログイン時にはget startedだけを表示し、プロフィール編集画面への遷移も可能に

Commit:
ヘッダーのlink_toタグへの変更

Commit:
first form_for!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Commit:
form_forメソッドをクラスに適用し、password・nickname・emailといったDBに保存を必要とするinputタグをtext_fieldに書き換えてビューは適用されたがDBに値が保存されない…
nicknameはdeviseのデフォルトではないのでconfigure_permitted_parametersメソッドを当ててみたけどそもそもDBが応答してくれないので働いてるかわからない。

Commit:
pry-railsの実装、登録後のホーム画面への遷移

Commit:
nicknameの表示、ログアウト機能実装、サインアップ後ホーム画面へ遷移
以上でセッション周り。

hc0208 commented on app/controllers/application_controller.rb:
```diff
@@ -1,5 +1,13 @@
+   before_action :configure_permitted_parameters, if: :devise_controller?
+
+    def configure_permitted_parameters
+      # binding.pry
```
binding.pryは消しておきましょう！

hc0208 commented on config/routes.rb:
```diff
@@ -1,7 +1,20 @@
+
+  # devise_scope :user do
+  #   root :to => 'devise/registrations#create'
+  # end
```
ここのコメントアウトもお願いします。

hc0208 commented:
LGTM
[![LGTM](http://lgtm.in/p/sYRWqzmnq)](http://lgtm.in/i/sYRWqzmnq)

Commit:
delete comment out
